cluster/src/main/scala/io/snappydata/impl/LeadImpl.scala

```diff
@@ -124,6 +125,15 @@ class LeadImpl extends ServerImpl with Lead with Logging {
         conf.set(key, v)
       })
 
+      /**
+       * If kryoserializer is being used then register snappyclasses for kryoserializer.
+       * The snappyclasses will not get registered if user is creating its own spark context.
+       * In that case user has to specify the snappy kryo registrator externally
+       */
+      if (conf.contains("spark.serializer") && conf.get("spark.serializer").equals("org.apache.spark.serializer.KryoSerializer")) {
+        conf.set("spark.kryo.registrator", getSnappyKryoRegistrarClass())
```

Comment:
What happens when user also specifies any registrator ? Can we have multiple registrator ? I suppose not. Can we not use spark.kryo.classesToRegister spark conf property. This will enable us to only append our classes. We can keep a list of classes we intent to register in one place. Just read the conf property and append our list to it.